### DISCUSSION TITLE
How to permanently delete the experiment?

### DISCUSSION THREAD

**Discussion by PeterPirog** [mlflow/mlflow#6293] [Q&A]
How to permanently delete the experiment?
I get an error if I try create again experiment with the same name even when I deleted previous version:
`mlflow.exceptions.MlflowException: Cannot set a deleted experiment 'experiment1' as the active experiment. You can restore the experiment, or permanently delete the experiment to create a new one.`

If there is no command to delete permanently where are stored files to delete its manually?
I try to use command [mlflow gc](https://www.mlflow.org/docs/latest/cli.html#mlflow-gc) after delete experiment by command [mlflow experiments delete -x 1](https://www.mlflow.org/docs/latest/cli.html#mlflow-experiments-delete) but it doesn't work.

**Comment by dagap** [comment]
Is  there  any progress on this front? it is extremely frustrating thatit should be so difficult to permnently delete experiments in SQL backend MLFlow instances.

The API call for `delete_experiment` cannot delete the experiments which have already been marked as deleted (from the front end for example). `mlflow gc` does not not do anything for SQL backend instances either.

**Comment by zmunro** [comment]
This is actually insane that this hasn't been fixed yet.

**Comment by B-Step62** [comment]
Refer to this issue: https://github.com/mlflow/mlflow/issues/12947#issuecomment-2292594760

TLDR; - please set `--backend-store-uri` when running `mlflow gc`, just like how you start the tracking server.
